### ISSUE TITLE
No strict deep equal option

### ISSUE THREAD

**Issue by nathanboktae** [visionmedia/supertest#256]
I was very surprised to change some of my APIs to stop returning quoted strings of numbers but instead return actual numbers and have all my tests still pass. Since supertest is using `assert.deepEqual`, it only does coercive (`==`) equals and not strict equal (`===`).

Should supertest switch to use a deep equal implementation that is strict equals? Perhaps provide an option to specify this? how would it look?

```javascript
      request
        .get('/heathcheck')
        .expect(200)
        .expect({
          node_count: 3
        }, { strict: true })
        .end(done)
```

or

```javascript
      request
        .get('/heathcheck')
        .expect(200)
        .expect([{
          node_count: 3
        }])
        .end(done, { strict: true })
```

I like the former. I'd be glad to put a PR for this.

**Comment by raine** [comment]
https://github.com/ramda/ramda/pull/1409#discussion_r40562734

**Comment by raine** [comment]
Is there any reason why it should be asserted with a `deepEqual` implementation that uses non-strict equals?

**Comment by nathanboktae** [comment]
> Is there any reason why it should be asserted with a deepEqual implementation that uses non-strict equals?

Exactly I don't see any reason you would want non-strict equals, as your HTTP contract should be strict.

**Comment by nathanboktae** [comment]
@mikelax Any reason not to use a strict deepEquals? the fact that you could write contract tests with supertest, have a regression where you are returning quoted numbers and your tests pass but your HTTP contract broken means that supertest failed at its one job.

**Comment by nathanboktae** [comment]
This appears fixed in `0.15.0` - I started to have test failures like this:

```
AssertionError: expected { Object (user_id, _first_event_at, ...) } to deeply equal { Object (user_id, _first_event_at, ...) }
      + expected - actual

         "friends": 3
         "gold_balance": 2.5
         "hit_rate": 3.4
      -  "hitpoints": 4000
      +  "hitpoints": "4000"
         "kill_rate": 1.3
         "level": 13
```

Thanks!
